cursorless.nvim/test/nvim-shim.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+export XDG_CONFIG_HOME='test/xdg/config/'
+export XDG_STATE_HOME='test/xdg/local/state/'
+export XDG_DATA_HOME='test/xdg/local/share/'
+
+# FIXME: For the flake.nix devshell I plan to eventually just install the unwrapped binary straight up so I can
+# reference it for these tests, but until then...
+function resolve_nvim_path() {
+  nvim=$(type -p "nvim" | awk '{print $NF}')
+  if file "${nvim}" | grep -q 'executable'; then
+    echo "${nvim}"
+    return
+  fi
+
+  # On NixOS, this might still be a wrapper, so we need to make sure we find the real binary
+  current_link="${nvim}"
+  while [ -L "${current_link}" ]; do
+    current_link=$(readlink "${current_link}")
+  done
+  nvim="${current_link}"
+
+  if file "${current_link}" | grep -q 'script'; then
+    nvim=$(grep 'exec -a' "${current_link}" | cut -f4 -d" " | tr -d '"')
+  fi
+  echo "${nvim}"
+}
+
+mkdir -p ${XDG_CONFIG_HOME} ${XDG_STATE_HOME} ${XDG_DATA_HOME}
+mkdir -p ${XDG_DATA_HOME}/nvim/site/pack/testing/start || true
+link_path="${XDG_DATA_HOME}/nvim/site/pack/testing/start/cursorless.nvim"
+if [ ! -e "${link_path}" ]; then
+  # FIXME: Expects to be in cursorless.nvim, so we should force cd here?
+  ln -sf "${PWD}" ${link_path}
+fi
+
+NVIM=$(resolve_nvim_path)
```

Comment:
If I use the flake.nix I used from https://github.com/saidelike/cursorless/pull/8 where I install bare nvim and force it to the beginning of PATH, I think I can skip the path resolution here. Will still need to invoke with `command nvim` though, but will simplify this a bunch.